pkg/webhooks/jobset_webhook.go

```diff
@@ -390,3 +396,21 @@ func replicatedJobByName(js *jobset.JobSet, replicatedJob string) *jobset.Replic
 	}
 	return nil
 }
+
+// validateReplicatedJobsUpdate validates the updates for elastic jobs
+// Changing length of jobs, name of jobs and the templates are forbidden
+func validateReplicatedJobsUpdate(currentRepJobs, oldRepJobs []jobset.ReplicatedJob) error {
+	// Changing length of replicated jobs on updates is forbidden
+	if len(currentRepJobs) != len(oldRepJobs) {
+		return fmt.Errorf("updates can not change the length of replicated jobs")
+	}
+	for i := range currentRepJobs {
+		if currentRepJobs[i].Name != oldRepJobs[i].Name {
+			return fmt.Errorf("updates can not change job names or reorder the jobs")
+		}
+		if !equality.Semantic.DeepEqual(currentRepJobs[i].Template, oldRepJobs[i].Template) {
```

Comment:
Will this conflict with the changes in #579 allowing mutability of pod template specs if the JobSet is suspended?

Reply:
I don’t think so but if we want the Kueue integration I’m happy to let @mimowo work go first and I’ll rebase.

Reply:
Didn't see any issue here.